src/System.Private.ServiceModel/tools/scripts/StopWcfSelfHostedSvcIfRunning.cmd

```diff
@@ -0,0 +1,67 @@
+@if "%_echo%" neq "on" echo off
+setlocal
+
+REM This CMD stops the Self hosted WCF test service.
+REM It will self-elevate to kill the local process.
+
+set _setupMonitorFile=%~dp0..\..\..\..\SelfHostedWcfServiceMonitor.log
+
+REM Detect if the process is running locally
+tasklist /FI "imagename eq SelfHostedWCFService.exe" | findstr /i /C:"SelfHostedWCFService.exe" 1>nul
+if [%ERRORLEVEL%] == [0] goto stopService
+ 
+REM If it's not running, exit with 0 to indicate nothing was done  
+echo Detected Self hosted WCF Test Service is not running.
+exit /b 0
```

Comment:
Recommend to follow the common coding style as all existing scripts - all code path has a common exit at the end of script.